ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/validator/reading/StateReader.scala

```diff
@@ -20,7 +20,7 @@ trait StateReader[Key, Value] {
     * @param keys list of keys to look up
     * @return values corresponding to the requested keys, in the same order as requested
     */
-  def read(keys: Seq[Key])(implicit executionContext: ExecutionContext): Future[Seq[Value]]
+  def read(keys: Iterable[Key])(implicit executionContext: ExecutionContext): Future[Seq[Value]]
```

Comment:
I changed this to allow the keys to be a `Set` or a key set, and kept it because we now pass an iterable to this function.

We now store the read set as a `Map[DamlStateKey, Fingerprint]`, rather than a `Seq[(Key, Fingerprint)]`. To perform conflict detection, we need to get the keys and values separately, and `Map#unzip` returns `(Iterable[K], Iterable[V])`, so to avoid converting to a `Seq`, we need to be able to accept an iterable.

This now aligns with `LedgerStateOperations#readState`, which accepts an `Iterable[Key]` and returns a `Seq[Option[Value]]`.